optuna/pruners/_hyperband.py

```diff
@@ -284,12 +284,14 @@ class _BracketStudy(optuna.study.Study):
                 "_study",
             )
 
-            def __init__(self, study: "optuna.study.Study", bracket_id: int) -> None:
+            def __init__(
+                self, study: "optuna.study.Study", pruner: BasePruner, bracket_id: int
```

Comment:
Since this class is instantiated by using `HyperbandPruner`, I suppose that we can make the type hint stronger.  What do you think about it? Note that mypy was happy to replace `basePruner` with `HyperbandPruner` on my local machine.

```suggestion
                self, study: "optuna.study.Study", pruner: HyperbandPruner, bracket_id: int
```